app/src/androidTest/java/org/mozilla/tv/firefox/ui/PageLoadTest.kt

```diff
@@ -1,89 +1,43 @@
+
+/**
+ * Tests that the browser can successfully load pages from the network. This works with our other UI tests
+ * which should load content from the disk to make them less fragile.
+ *
+ * This test loads a few pages off the network and verifies their content to ensure they're actually loaded.
+ * Verification is minimal to reduce the chance of the test breaking from the web page changing.
+ */
 class PageLoadTest {
 
-    private val mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
-
-    @Rule @JvmField var mActivityTestRule: ActivityTestRule<MainActivity> = MainActivityTestRule()
-
-    @After
-    fun tearDown() {
-        mActivityTestRule.activity.finishAndRemoveTask()
-    }
+    @Rule @JvmField val activityTestRule = MainActivityTestRule()
 
+    /* ktlint-disable no-blank-line-before-rbrace */ // This imposes unreadable grouping.
     @Test
-    fun PageLoadTest() {
-        onView(ViewMatchers.withId(R.id.tileContainer))
-            .perform(RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(TILE_POSITION, click()))
-
-        onView(ViewMatchers.withId(R.id.webview))
-            .check(matches(isDisplayed()))
-
-        onWebView()
-            .withElement(findElement(Locator.ID, TILE_WEBSITE_ELEMENT))
-
-        mDevice.pressMenu()
-
-        onView(ViewMatchers.withId(R.id.navUrlInput))
-            .check(matches(isDisplayed()))
-            .check(matches(withText(containsString("google"))))
-
-        onView(allOf<View>(withId(R.id.navUrlInput), isDisplayed(), hasFocus()))
-            .perform(replaceText(MOZILLA_URL))
-            .perform(pressImeActionButton())
-
-        onView(ViewMatchers.withId(R.id.webview))
-            .check(matches(isDisplayed()))
-
-        onWebView()
-            .withElement(findElement(Locator.CSS_SELECTOR, MOZILLA_PAGE_ELEMENT))
-
-        mDevice.pressMenu()
-
-        onView(ViewMatchers.withId(R.id.navUrlInput))
-            .check(matches(isDisplayed()))
-            .check(matches(withText(containsString("mozilla"))))
-    }
-
-    companion object {
-        private val TILE_POSITION = 1 // Google Video Search
-        private val TILE_WEBSITE_ELEMENT = "hplogo" // Google logo
-        private val MOZILLA_URL = "mozilla.org"
-        private val MOZILLA_PAGE_ELEMENT = ".primary-title"
+    fun networkPageLoadTest() {
+        val googleUrl = "https://google.com".toUri()!!
+        val mozillaUrl = "https://mozilla.org".toUri()!!
+
+        navigationOverlay { }.enterUrlAndEnterToBrowser(googleUrl) {
+            assertDOMElementExists(Locator.ID, "hplogo") // google logo
+        }.openOverlay {
+            assertURLBarTextContains("google")
+
+        }.enterUrlAndEnterToBrowser(mozillaUrl) {
+            assertDOMElementExists(Locator.CLASS_NAME, "mzp-c-navigation-logo") // mozilla logo
+        }.openOverlay {
+//            assertURLBarTextContains("mozilla") // TODO: on emulators, url is not always the website just visited. (#1444)
```

Comment:
It would be possible to work around this by disabling turbo mode.  This problem is only present on emulators, so that may be acceptable. 🤷‍♂️

Reply:
@Baron-Severin Good idea: I ended up disabling turbo mode (adding a comment about the workaround) and uncommenting this assertion. This required a minor refactor of Matchers.kt, which you may be interested in looking at but I'm not concerned so I didn't request another review.